src/trunk/libs/seiscomp3/io/recordstream/sdsarchive.cpp

```diff
@@ -441,6 +442,42 @@ bool SDSArchive::isEnd() {
 
 
 
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+void SDSArchive::setupRoots() {
+	if (!_roots.empty()) return;
+	
+	size_t pold, p; 
+	pold = p = 0;
+
+	while (p != string::npos) {
+		pold = p;
+		p = _arcroot.find(",", pold);
+		
+		std::string source = _arcroot.substr(pold, p-pold);
+		_roots.push_back(source);
+		
+		SEISCOMP_DEBUG("+ Add to root list: %s", source.c_str());
+		
+		if (p != string::npos && _arcroot[p] == ',') p=p+1;
+	}
+	SEISCOMP_DEBUG("Total of %ld roots are in use.", _roots.size());
+	
+	return;
+}
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```

Comment:
This function can be replaced by `Core::split(_roots, _arcroot, ",")`

In fact I would suggest to

- convert `_arcroot` in to a vector and rename it to `_arcroots`
- initialize it using `Core::split` in `setSource`
- remove member `_roots`

Reply:
Suggested changes would make code more usable and reliable to maintain but has some impacts on implemented features.

SDSArchive RecordStream implements a constructor from another SDSArchive object that uses the archiveRoot() method. The archiveRoot() originally returns a string that is now a list. Another use of the archiveRoot() is inside the '=' operator that is used to make two instances of SDS pointing to the same archive. The archiveRoot() is marked as public. 

Looking at sc3 code I could not find a place where archiveRoot() is used, looks like a lava flow code. We have the chance now to remove it, cleaning up unused features or, re-factor it to return  list but had no guaranties that we will not break other parts of codes or 3rd part plug-ins since this was a public feature.

What is the best alternative? Refactor and keep this apparent lava flow, or clean up, removing constructor from another instance and '=' operator and dependencies?